source/apiVolontaria/volunteer/views.py

```diff
@@ -278,3 +279,127 @@ def delete(self, request, *args, **kwargs):
             'detail': "You are not authorized to delete an event.",
         }
         return Response(content, status=status.HTTP_403_FORBIDDEN)
+
+
+class Participations(generics.ListCreateAPIView):
+
+    """
+
+    This view handles the link between User and Events (participations)
+
+    get:
+    Return a list of all the existing Participations.
+
+    post:
+    Create a new Participation.
+
+    """
+
+    serializer_class = serializers.ParticipationBasicSerializer
+
+    def get_queryset(self):
+        if self.request.user.is_superuser:
+            return models.Participation.objects.all()
+        elif self.request.user.has_perm('volunteer.add_participation'):
+            queryset = models.Participation.objects.all()
+
+            list_exclude = list()
+            for participation in queryset:
+                if participation.user != self.request.user:
+                    list_exclude.append(participation)
+
+            queryset = queryset.\
+                exclude(
+                    pk__in=[participation.pk for participation in list_exclude]
+                )
+
+            return queryset
+
+    def post(self, request, *args, **kwargs):
+        if self.request.user.is_superuser:
+            return self.create(request, *args, **kwargs)
+        elif self.request.user.has_perm('volunteer.add_participation'):
+            if request.data['user'] == self.request.user.id:
+                return self.create(request, *args, **kwargs)
+
+            content = {
+                'detail': "Invalid user id.",
+            }
+            return Response(content, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Good job! It's a good check 👍 

I think the 400 is not the best code, i think it's more a 403. 

 - You're authorized (connected, token ok)
 - Your request is good (arguments name ok, required ok)
 - You can't do that! It's not allowed because you don't have some permissions like a superuser

So for me it's a 403 and not a bad request.